### DISCUSSION TITLE
NullPointerException : Generate Receiver Point using delaunay grid

### DISCUSSION THREAD

**Discussion by yslidet** [Universite-Gustave-Eiffel/NoiseModelling#354] [Q&A] [answered]
Hi everyone, 

I have been trying to generate noise propagation from the road according to the [Noise Map from OSM - Tutorial](https://noisemodelling.readthedocs.io/en/latest/Noise_Map_From_OSM_Tutorial.html).

Data used:  [map.osm](https://entuedu-my.sharepoint.com/:u:/g/personal/seanglidet_yean_staff_main_ntu_edu_sg/EQQdYrsYoetNkIWw75b9iEkBsVxfYH_cxpbUHehYh1QL-A?e=Ogv937) with **targetSRID=4326**

However, having selected BUILDINGS and ROADS as the required inputs, I have gotten the errors as the following 
WPS console: 
`Process failed during execution javax.script.ScriptException: java.lang.NullPointerException java.lang.NullPointerException`

Terminal - Running Server Code:
`DEBUG [geoserver.wps] - Saving status ExecutionStatus [processName=Receivers:Delaunay_Grid, executionId=d39509fd-e91d-422e-b8d0-09669d4a6e9e, asynchronous=false, phase=FAILED, progress=0.0, userName=anonymous, creationTime=Thu Jun 17 10:42:10 SGT 2021, completionTime=Thu Jun 17 10:42:10 SGT 2021, lastUpdated=Thu Jun 17 10:42:10 SGT 2021, expirationDate=Thu Jun 17 10:42:10 SGT 2021, estimatedCompletion=null, nextPoll=Thu Jun 17 10:42:10 SGT 2021, task=null, exception=org.geotools.process.ProcessException: javax.script.ScriptException: java.lang.NullPointerException, nodeId=192.168.1.96]
17 Jun 10:42:10 INFO [geoserver.wps] - 
Request: execute
    service = WPS
    version = 1.0.0
    baseUrl = http://localhost:9580/geoserver/
    identifier:
        value = Receivers:Delaunay_Grid
    dataInputs:
        input[0]:
            identifier = net.opengis.ows11.impl.CodeTypeImpl@1332b8bb (value: sourcesTableName, codeSpace: null)
            data = net.opengis.wps10.impl.DataTypeImpl@f5866f7input[0]:
            identifier = net.opengis.ows11.impl.CodeTypeImpl@4b27f4a8 (value: tableBuilding, codeSpace: null)
            data = net.opengis.wps10.impl.DataTypeImpl@46b29cde
    responseForm:
        rawDataOutput:
            identifier = net.opengis.ows11.impl.CodeTypeImpl@404313d3 (value: result, codeSpace: null)
`

Really appreciate your help and looking forward to your diagnosis for what cause this error. 

It is to be noted that I did run with [another location's OSM](https://entuedu-my.sharepoint.com/:u:/g/personal/seanglidet_yean_staff_main_ntu_edu_sg/Ed3NiUZDOANHjtKyAUxy4C0BxdW0o3JgvChi25h91KmTrA?e=7MP7bM)  and there was no error.

**Comment by pierromond** [comment, marked as answer]
We didn't have time to update the tutorial, but please prefer import_osm_pbf, using a pbf file extracter from BBBike. It works better ;)

As it is an issue, could you please answer to this template ?

**NoiseModelling version**
Version number of NoiseModelling you are using (ex 3.2)

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Geoserver log**

Attach geoserver log located in **data_dir\logs\geoserver.log**

**Expected behavior**
A clear and concise description of what you expected to happen.

**Reply by yslidet** [reply]
As I am using the NoiseModelling version 3.3.1, I was not able to find the "import_osm_pbf" in the WPS process builder. May I know if I was looking at the right place? 

Regarding the issues, 
NoiseModelling version: 3.3.1

Describe the bug
- Fail to generate delaunay grid of receivers using BUILDINGS and ROADS tables, imported from map.osm. 
- `Process failed during execution javax.script.ScriptException: java.lang.NullPointerException java.lang.NullPointerException`

To Reproduce
- Go to 'NoiseModelling_3.3.1' folder. It could be downloaded [here](https://github.com/Ifsttar/NoiseModelling/releases/tag/v3.3.1)
- Run `bash bin/startup.sh` 
- Go to web browser, http://localhost:9580
- Run WPS **Clean_Database**
- Run WPS **Import_OSM** by inputing pathFile = [path/to/map.osm](https://entuedu-my.sharepoint.com/personal/seanglidet_yean_staff_main_ntu_edu_sg/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fseanglidet%5Fyean%5Fstaff%5Fmain%5Fntu%5Fedu%5Fsg%2FDocuments%2F3%2E0%2ENS%2DTip%2FScript%5FNoiseMod%5FV1%2FCASE%5FSTUDY%2FClementiWalk%5FNoiseCapture%2Finputs%2Fclementi%5Fwalk%2Eosm&parent=%2Fpersonal%2Fseanglidet%5Fyean%5Fstaff%5Fmain%5Fntu%5Fedu%5Fsg%2FDocuments%2F3%2E0%2ENS%2DTip%2FScript%5FNoiseMod%5FV1%2FCASE%5FSTUDY%2FClementiWalk%5FNoiseCapture%2Finputs&originalPath=aHR0cHM6Ly9lbnR1ZWR1LW15LnNoYXJlcG9pbnQuY29tLzp1Oi9nL3BlcnNvbmFsL3NlYW5nbGlkZXRfeWVhbl9zdGFmZl9tYWluX250dV9lZHVfc2cvRVFRZFlyc1lvZXROa0lXdzc1YjlpRWtCc1Z4ZllIX2N4cGJVSGVoWWgxUUwtQT9ydGltZT1pVGtNM2hnMjJVZw), targetSRID = 4326. 
- Run WPS **Display_Database**, results in BUILDINGS, ROADS, GROUND, MAP_BUILDINGS, TMP_RELATION_BUILDINGS_BUILDINGS
- Run WPS **Delaunay_Grid** by inputing tableBuilding='BUILDINGS', sourcesTableName='ROADS'
- See Error: 
`Process failed during execution javax.script.ScriptException: java.lang.NullPointerException java.lang.NullPointerException`

Attach geoserver log located in data_dir\logs\
- [geoserver.log](https://entuedu-my.sharepoint.com/:u:/g/personal/seanglidet_yean_staff_main_ntu_edu_sg/EePCyBIUM9hJsSnM5DMeAxIBKpQyUVLbJVaKn7es1cyA7g?e=4H3zb8)
- [geoserver.log.1](https://entuedu-my.sharepoint.com/:u:/g/personal/seanglidet_yean_staff_main_ntu_edu_sg/EUt2LG4-PV1MneIMcgFjKkwBFVRI5uQV5qCWfPbaWYp5_A?e=l2F0Lh)

Expected behavior
Successfully created RECEIVERS table.

**Reply by nicolas-f** [reply]
Hi,
This is an issue with buildings envelope. Maybe check buildings table and use the last version of NoiseModelling.

**Reply by yslidet** [reply]
Hi Nicolas, 

I have tried the latest version as you suggested (v3.4.1). However, the issues of generating the Delaunay_Grid mentioned earlier remain. I have tried checking the BUILDINGS generated from map.osm with QGIS as well. **_I am not sure what could be the issue._** 

In addition, I have tried to generate the grid from other WPS blocks (e.g. BUILDING_Grid) and it seems alright. I proceed to try Road_Emission_from_Traffic by inputing the ROADS (generated from map.osm) and I obtained this errors: 
`Process failed during execution javax.script.ScriptException: java.lang.IllegalArgumentException: Error : Please use a metric projection for ROADS. java.lang.IllegalArgumentException: Error : Please use a metric projection for ROADS. Error : Please use a metric projection for ROADS.`

**Reply by nicolas-f** [reply]
Hi.

As stated in the tutorial you can't use 4326 coordinate system for creating a noisemap.

https://noisemodelling.readthedocs.io/en/latest/Tutorials_FAQ.html#metric-srid

**Reply by pierromond** [reply]
Hi ! You can use the WPS block change_srid to convert from 4326 to your chosen metric system.

**Reply by yslidet** [reply]
Dear @nicolas-f and @pierromond , thank you so much for your kind assistant and my apology for the late responses. The issue is fixed as you mentioned that it was the coordinate system.
